src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -767,9 +784,10 @@ public static void EmitObject(string objectFilePath, IEnumerable<DependencyNode>
                 }
                 else
                 {
-                    objectWriter.SetSection(MethodCodeNode.UnixContentSection);
+                    managedCodeSection = MethodCodeNode.UnixContentSection;
                     objectWriter.SetSection(LsdaSection);
                 }
+                objectWriter.SetCodeSectionAttribute(managedCodeSection);
```

Comment:
Is it necessary to add a new ObjectWriter API for this?  We have the existing `CustomSectionAttributes` flags we pass through when setting a section that objectwriter could look for. We could key off `CustomSectionAttributes.Executable` for instance.